storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -182,13 +182,12 @@ private CompletableFuture<Void> storeRemoteLogMetadata(TopicIdPartition topicIdP
             CompletableFuture<RecordMetadata> produceFuture = producerManager.publishMessage(remoteLogMetadata);
 
             // Create and return a `CompletableFuture` instance which completes when the consumer is caught up with the produced record's offset.
-            return produceFuture.thenApplyAsync(recordMetadata -> {
+            return produceFuture.thenAcceptAsync(recordMetadata -> {
                 try {
-                    consumerManager.waitTillConsumptionCatchesUp(recordMetadata);
+                    consumerManager.waitTillConsumptionCatchesUp(recordMetadata.partition(), recordMetadata.offset());
```

Comment:
We should pass `recordMetadata` only, right?

Reply:
Of course, missed this one -- pushed fixes.